### PR TITLE
ROX-18247: Add env var for registry mirroring support

### PR DESCRIPTION
**Pull request by dcaravel**
## Description

Adds an environment variable `ROX_REGISTRY_MIRRORING_ENABLED` that will be used to control enabling / disabling support for registry mirroring capabilities.

Is un-used as of this PR, will be consumed as the stories in ROX-11347 are implemented.

## Checklist
- [x] Investigated and inspected CI test results
- [ ] ~Unit test and regression tests added~
- [x] Evaluated and added CHANGELOG entry if required
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

Unit tests and docs will be added as part of stories mentioned in ROX-11347

## Testing Performed

N/A

### CODE REVIEW

File: pkg/env/registry_mirroring.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package env
+
+var (
+	// RegistryMirroringEnabled enables processing registry mirrors during image enrichment.
+	RegistryMirroringEnabled = RegisterBooleanSetting("ROX_REGISTRY_MIRRORING_ENABLED", false)
```
Comment (RTann) [inline pkg/env/registry_mirroring.go:5]: Do we want this to be a permanent setting or just a feature flag?

Comment (dcaravel) [reply pkg/env/registry_mirroring.go:5]: Immediately act as 'feature flag' but if it sticks around long enough permanent 😉 . Will observe how the feature behaves over time.

Prefer not to use the `features` package due to that value not being changeable post release.

Comment (RTann) [reply pkg/env/registry_mirroring.go:5]: if you are sure this is shipping in the next release, then this is good. if it misses, then we'll switch it to a feature flag

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 ### Added Features
 
 - Telemetry collection enabled by default for self-managed installations. Opt-out is available on bundle generation, or at any time via the System Configuration UI.
+- A new environment variable `ROX_REGISTRY_MIRRORING_ENABLED` has been added that when set to `true` will enable processing registry mirrors during image enrichment.
```
Comment (RTann) [inline CHANGELOG.md:13]: specifically as specified in OpenShift via the ImageContentSourcePolicy, right?

**Review by RTann — APPROVED** [review]
Approving, but just want to clarify this is just supporting the ImageContentSourcePolicy and similar OpenShift policies, specifically, right? If so, let's mention it in the CHANGELOG

File: CHANGELOG.md (hunk 2 above)
Comment (dcaravel) [reply CHANGELOG.md:13]: Yes OpenShift - will also include `ImageDigestMirrorSet` and `ImageTagMirrorSet`

**Pull request merged**